ts/src/bingx.ts

```diff
@@ -0,0 +1,2721 @@
+            'marginMode': marginMode,
+            'liquidationPrice': undefined,
+            'entryPrice': this.safeNumber (position, 'avgPrice'),
+            'unrealizedPnl': this.safeNumber (position, 'unrealizedProfit'),
+            'percentage': undefined,
+            'contracts': undefined,
+            'contractSize': undefined,
+            'markPrice': undefined,
+            'lastPrice': undefined,
+            'side': this.safeStringLower (position, 'positionSide'),
+            'hedged': undefined,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'lastUpdateTimestamp': undefined,
+            'maintenanceMargin': undefined,
+            'maintenanceMarginPercentage': undefined,
+            'collateral': this.safeString (position, 'positionAmt'),
+            'initialMargin': this.safeNumber (position, 'initialMargin'),
+            'initialMarginPercentage': undefined,
+            'leverage': this.safeNumber (position, 'leverage'),
+            'marginRatio': undefined,
+        });
+    }
+
+    async createOrder (symbol: string, type, side: OrderSide, amount, price = undefined, params = {}) {
+        /**
+         * @method
+         * @name bingx#createOrder
+         * @description create a trade order
+         * @see https://bingx-api.github.io/docs/#/spot/trade-api.html#Create%20an%20Order
+         * @see https://bingx-api.github.io/docs/#/swapV2/trade-api.html#Trade%20order
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} type 'market' or 'limit'
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} amount how much of currency you want to trade in units of base currency
+         * @param {float|undefined} price the price at which the order is to be fullfilled, in units of the quote currency, ignored in market orders
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        let response = undefined;
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('createOrder', market, params);
+        type = type.toUpperCase ();
+        const request = {
+            'symbol': market['id'],
+            'type': type,
+            'side': side.toUpperCase (),
+        };
+        const isMarketOrder = type === 'MARKET';
+        const stopPriceRaw = this.safeValue2 (params, 'stopPrice', 'triggerPrice');
+        let stopPrice = undefined;
+        if (stopPriceRaw !== undefined) {
+            stopPrice = this.priceToPrecision (symbol, stopPriceRaw);
+        }
+        const stopLossPrice = this.safeValue (params, 'stopLossPrice');
+        const takeProfitPrice = this.safeValue (params, 'takeProfitPrice');
```

Comment:
@AlgoQ I think we're missing the type 1 stop-orders unification 
![image](https://github.com/ccxt/ccxt/assets/43336371/60f18d55-88f7-4cfd-b587-a8c8eced7adf)